content/_code-samples/escrow/js/cancel-escrow.js

```diff
@@ -1,40 +1,94 @@
+    var xrpl = require('xrpl')
+  }
+  // Construct an EscrowCancel transaction to cancel an Escrow which has passed its CancelAfter due date
+  // https://xrpl.org/escrow.html#escrow
+  // https://xrpl.org/escrowcancel.html#escrowcancel
 
-const myAddr = 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn'
-const mySecret = 's████████████████████████████'
+  async function main() {
+    // To cancel an Escrow, you'd need to have an account which has sent an Escrow with a 'CancelAfter' field pass its time
+    // You could use 'create-escrow.js' on the same folder to import an account
 
-const myEscrowCancellation = {
-  owner: myAddr,
-  escrowSequence: 366
-}
-const myInstructions = {
-  maxLedgerVersionOffset: 5
-}
+    seed = ""
+
+    // Connect to a testnet node
+    console.log("Connecting to Testnet...")
+    const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
+    await client.connect()
+
+    wallet = xrpl.Wallet.fromSeed(seed)
+
+    console.log(" Sending Account:", wallet.address)
+    console.log("            Seed:", wallet.seed)
+
+    // Query the account's objects (Type: Escrow)
+    const response = await client.request({
+        "command": "account_objects",
+        "account": wallet.address,
+        "ledger_index": "validated",
+        "type": "escrow"
+    })
+
+    // Only append Escrows that were sent by this Account, incoming Escrows are ignored since we have no authority to cancel them
+    var outgoing = []
+    for (var i = 0; i < response.result.account_objects.length; i++) {
+        if (response.result.account_objects[i].Account == wallet.address) {
+            outgoing.push(response.result.account_objects[i])
+        }
+    }
+
+    // Get current datetime
+    const date = new Date();
+
+    // Only append outgoing Escrows that have passed their CancelAfter time
+    var outgoing_cancelable = []
+    for (var i = 0; i < outgoing.length; i++) {
+        const CancelAfter_date = new Date(xrpl.rippleTimeToISOTime(outgoing[i].CancelAfter));
+        if (CancelAfter_date.getTime() <= date.getTime()) {
+            outgoing_cancelable.push(outgoing[i])
+        }
+    }
+    
+    if (outgoing_cancelable.length != 0 ) {
+        console.log("\nOutgoing/Sent escrow(s) that have passed their CancelAfter time:")
+        for (var i = 0; i < outgoing_cancelable.length; i++) {
+            console.log(`\n${i+1}. Index (ObjectID/keylet): ${outgoing_cancelable[i].index}`)
+            console.log(` - Account: ${outgoing_cancelable[i].Account})`)
+            console.log(` - Destination: ${outgoing_cancelable[i].Destination}`)
+            console.log(` - Amount: ${outgoing_cancelable[i].Amount} drops`)
+            console.log(` - PreviousTxnID: ${outgoing_cancelable[i].PreviousTxnID}`)
+        }
+
+        console.log(`\nWe're going to cancel Escrow ${outgoing_cancelable[0].index} which was destined for ${outgoing_cancelable[0].Destination} @ ${outgoing_cancelable[0].Amount} drops...`)
```

Comment:
Same suggestion as for FinishEscrow

```suggestion
        console.log(`\nWe're going to cancel Escrow ${outgoing_cancelable[0].index} which was sending ${outgoing_cancelable[0].Amount} drops to ${outgoing_cancelable[0].Destination}`)
```